tests/swats_test.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file swats_test.cpp
+ * @author Marcus Edel
+ *
+ * Test file for the SWATS optimizer.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <ensmallen.hpp>
+#include "catch.hpp"
+#include "test_function_tools.hpp"
+
+using namespace ens;
+using namespace ens::test;
+
+/**
+ * Test the SWATS optimizer on the simple SGD function.
+ */
+TEST_CASE("SWATSSGDFunction","[SWATSTest]")
+{
+  SGDTestFunction f;
+  SWATS optimizer(1e-3, 1, 0.9, 0.999, 1e-6, 400000, 1e-9, true);
+
+  arma::mat coordinates = f.GetInitialPoint();
+  optimizer.Optimize(f, coordinates);
+
+  REQUIRE(coordinates[0] == Approx(0.0).margin(0.1));
+  REQUIRE(coordinates[1] == Approx(0.0).margin(0.1));
```

Comment:
Looks like this failed... maybe the test needs to be run multiple times and have at least one success?